tests/interfaces/test_batch_autograd.py

```diff
@@ -439,6 +439,38 @@ def cost(a, b, device):
         expected = [[-np.sin(a), 0], [np.sin(a) * np.sin(b), -np.cos(a) * np.cos(b)]]
         assert np.allclose(res, expected, atol=tol, rtol=0)
 
+    def test_tape_no_parameters(self, execute_kwargs, tol):
+        """Test that a tape with no parameters is correctly
+        ignored during the gradient computation"""
+        dev = qml.device("default.qubit", wires=1)
+
+        def cost(params):
+            with qml.tape.JacobianTape() as tape1:
+                qml.Hadamard(0)
+                qml.expval(qml.PauliX(0))
+
+            with qml.tape.JacobianTape() as tape2:
+                qml.RY(np.array(0.5, requires_grad=False), wires=0)
+                qml.expval(qml.PauliZ(0))
+
+            with qml.tape.JacobianTape() as tape3:
+                qml.RY(params[0], wires=0)
+                qml.RX(params[1], wires=0)
+                qml.expval(qml.PauliZ(0))
+
+            return sum(execute([tape1, tape2, tape3], dev, **execute_kwargs))
+
+        params = np.array([0.1, 0.2], requires_grad=True)
+        x, y = params
+
+        res = cost(params)
+        expected = 1 + np.cos(0.5) + np.cos(x) * np.cos(y)
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        grad = qml.grad(cost)(params)
+        expected = [-np.cos(y) * np.sin(x), -np.cos(x) * np.sin(y)]

```

Comment:
I'm not sure I understand this test, is the point that it is only differentiating w.r.t. the two inputs parameters and not all the gate parameters?

Reply:
Yep, essentially. It is testing the edge case

```python
>>> execute([tape1, tape2, tape3])
>>> tape1.trainable_params
set()  # empty because it has no parametrized gates
>>> tape2.trainable_params
set()  # empty because all params are requires_grad=False
```

Currently, in the master branch, this will fail for Torch and TensorFlow. This change fixes it:

![image](https://user-images.githubusercontent.com/2959003/132869246-d230b9ba-30ea-4acb-b446-6c4e69508539.png)

Reply:
I figure it is important to have a test for this edge case, to ensure it can't be broken in the future.